src/object/reply.c

```diff
@@ -81,10 +78,11 @@ void reply_pop(reply_t *reply, tcb_t *tcb)
             REPLY_PTR(prev_ptr)->replyNext = reply->replyNext;
             assert(call_stack_get_isHead(REPLY_PTR(prev_ptr)->replyNext));
         }
-
```

Comment:
Sorry for taking ages to get back to this. This would be really good to have for verification.

While we are here, I have one more change to ask. Would it be ok to swap the `if`-blocks just above here, so that we do the SC donation as the last thing before clearing the reply prev/next links? I'd imagine it won't matter which order they come, but is it actually the case? (@VictorPhan is doing the haskell change for this)

Reply:
@corlewis is this going to happen before this gets merged?

Reply:
@corlewis bump?

Reply:
Clarifying, the `if` with condition `tcb->tcbSchedContext == NULL` moves to after the `if` with the condition `prev_ptr != 0`

Reply:
Correct, and that change has now been added to this PR.